src/main/java/io/hgraphdb/models/VertexIndexModel.java

```diff
@@ -93,30 +97,26 @@ public void deleteVertexIndex(Vertex vertex, String key, Long ts) {
         ResultScanner scanner;
         try {
             scanner = table.getScanner(scan);
-            Iterator<Vertex> iterator = CloseableIteratorUtils.flatMap(
-                    CloseableIteratorUtils.concat(scanner.iterator(), IteratorUtils.of(Result.EMPTY_RESULT)),
+            List<Vertex> vertices = Lists.newArrayList(scanner.iterator())
```

Comment:
Please check if `graph.isParallelLoading()` and then retain the old code if `false`.